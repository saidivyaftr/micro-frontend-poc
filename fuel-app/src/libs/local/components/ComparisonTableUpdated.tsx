import { makeStyles } from '@material-ui/core/styles'
import { Button, ComparisonTable, Typography } from '@/shared-ui/components'
import { IComparison } from '@/shared-ui/components/ComparisonTable/types'
import { useMemo } from 'react'
import colors from '@/shared-ui/colors'
import InfoIconWhite from '@/shared-ui/react-icons/info-icon-white'
import { SITE_INTERACTION } from 'src/constants'
import { useRouter } from 'next/router'
const ComparisonTableUpdated = ({ data }: any) => {
  const classes = useStyles()
  const { title, items, legal, buttonText, buttonURL } = data

  const router = useRouter()
  const pageName = useMemo(() => {
    return router.asPath.split('/').splice(1, 3).join('/')
  }, [router])
  const list = useMemo((): IComparison[] => {
    const itemsData: IComparison[] = []
    items?.list?.map((listData: any) => {
      const header = listData?.headerDescription?.value || ''
      const headerLink = listData?.headerDescriptionLink?.url
      const logo = listData?.logo?.src || ''
      const properties = listData?.properties?.list?.map(
        ({ name, textValue, value, isPrimary, toolTip }: any) => {
          return {
            name: name?.value || '',
            textValue: textValue?.value,
            value: value?.value,
            isPrimary: !!isPrimary?.value,
            toolTip: toolTip?.value,
          }
        },
      )
      itemsData.push({ properties, header, headerLink, logo })
    })
    return itemsData
  }, [items])

  if (!data || Object.keys(data)?.length === 0) {
    return null
  }

  if (!data || Object.keys(data)?.length === 0) {
    return null
  }

  return (
    <div id="compare-fibers" className={classes.root}>
      <Typography tagType="h2" styleType="h3" className={classes.title}>
        {title?.value || ''}
      </Typography>
      <div className={classes.comparisonWrapper}>
        <ComparisonTable
          items={list}
          addBorderToHeader
          toolTipIcon={<InfoIconWhite />}
          hideBorderForTooltip={true}
          dropShadowForTooltip={true}
          rowHeaderstyleType="h6"
          styleModifier={{
            header: classes.header,
            textStyleType: 'p1',
            showRedCheckMarks: true,
            hidePreferredRowValue: true,
            tooltipStyles: classes.tooltipStyles,
            rowValueClassName: classes.tableRowClass,
            firstColumnStyles: classes.firstColumnStyles,
            valueTextCSS: classes.valueTextCSS,
          }}
        />
      </div>
      <Typography className={classes.legal} tagType="p" styleType="legal">
        {legal?.value || ''}
      </Typography>
      {buttonText?.value && (
        <div className={classes.cta}>
          <Button
            type="link"
            text={buttonText?.value}
            href={buttonURL?.url}
            triggerEvent={true}
            eventObj={{
              events: 'event14',
              eVar14: `${pageName}:${buttonText?.value?.toLowerCase()}`,
            }}
            interactionType={SITE_INTERACTION}
          />
        </div>
      )}
    </div>
  )
}

const useStyles = makeStyles(({ breakpoints }) => ({
  root: {
    maxWidth: 1200,
    margin: '84px auto 67px',
    [breakpoints.down('md')]: {
      padding: '0 16px',
    },
  },
  valueTextCSS: {
    fontSize: '1.125rem',
    [breakpoints.down('xs')]: {
      fontSize: '0.875rem',
      lineHeight: '1.125rem',
    },
  },
  comparisonWrapper: {
    marginTop: 64,
    borderTop: `1px solid ${colors.main.gray90}`,
    [breakpoints.down('md')]: {
      marginTop: 30,
    },
  },
  title: {
    textAlign: 'center',
  },
  header: {
    marginTop: '8px',
    '& img': {
      height: 25,
      [breakpoints.down('xs')]: {
        height: 20,
      },
    },
    [breakpoints.up('xs')]: {
      width: '33.33% !important',
      padding: '8px 16px',
    },
  },
  legal: {
    display: 'block',
  },
  cta: {
    display: 'flex',
    margin: '48px 0',
    marginBottom: '-12px',
    justifyContent: 'center',
  },
  tooltipStyles: {
    display: 'inline',
  },
  tableRowClass: {
    fontWeight: 400,
    [breakpoints.down('xs')]: {
      textAlign: 'center',
      padding: '8px 8px 32px 8px !important',
    },
    [breakpoints.up('xs')]: {
      width: '33.33% !important',
    },
  },
  firstColumnStyles: {
    width: '37%',
    [breakpoints.down('xs')]: {
      width: '100%',
      '& div': {
        padding: '1rem 0',
      },
    },
  },
}))

export default ComparisonTableUpdated
