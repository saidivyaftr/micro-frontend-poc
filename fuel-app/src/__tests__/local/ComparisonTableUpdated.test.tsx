import ComparisonTableUpdated from 'src/libs/local/components/ComparisonTableUpdated'
import { render, cleanup } from '@testing-library/react'

const MOCK_DATA = {
  title: {
    value: 'Comparison Table',
  },
  yesIcon: {
    value:
      'https://vsgstoqarg-539670-cdn-endpoint.azureedge.net/-/jssmedia/Project/Frontier/Dotcom/Images/shop/tv/check-mark-black.svg?rev=3361ced0fcec470ab7f821ebd997e697',
    alt: 'checkmark',
  },
  legal: {
    value: '',
  },
  buttonText: {
    value: '',
  },
  buttonURL: {
    url: '',
  },
  items: {
    list: [
      {
        headerDescription: {
          value: 'Header 1',
        },
        logo: {
          src: 'https://vsgstoqarg-539670-cdn-endpoint.azureedge.net/-/jssmedia/Project/Frontier/Dotcom/Images/fiber-2-gigabit-internet/logo-frontier.png?rev=78dfde7215fe416d80e38143dcebfa57',
        },
        iconColor: {
          color: {
            field: {
              value: '#ff0037',
            },
          },
        },
        properties: {
          list: [
            {
              name: {
                value: 'Speed',
              },
              toolTip: {
                value: 'Tooltip 1',
              },
              textValue: {
                value:
                  '<div> <span>2000 </span> Mbps download <br />\n<span>2000</span> Mbps upload</div>',
              },
              value: {
                value: false,
              },
              isPrimary: {
                value: false,
              },
            },
          ],
        },
      },
    ],
  },
}

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    asPath: '',
  }),
}))

describe('comparison table', () => {
  afterEach(cleanup)
  it('should render correctly', () => {
    const { asFragment, getByText } = render(
      <ComparisonTableUpdated data={MOCK_DATA} />,
    )
    expect(asFragment()).toMatchSnapshot()
    expect(getByText(MOCK_DATA.title.value)).toBeInTheDocument()
    expect(
      getByText(MOCK_DATA.items.list[0].properties.list[0].name.value),
    ).toBeInTheDocument()
  })

  it('should not render correctly', () => {
    const { queryByText } = render(<ComparisonTableUpdated data={''} />)
    expect(queryByText(MOCK_DATA.title.value)).not.toBeInTheDocument()
    expect(
      queryByText(MOCK_DATA.items.list[0].properties.list[0].name.value),
    ).not.toBeInTheDocument()
  })
})
