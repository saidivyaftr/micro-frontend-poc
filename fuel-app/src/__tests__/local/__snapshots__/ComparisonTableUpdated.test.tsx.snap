// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comparison table should render correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
    id="compare-fibers"
  >
    <h2
      class="makeStyles-title-3"
    >
      Comparison Table
    </h2>
    <div
      class="makeStyles-comparisonWrapper-2"
    >
      <div>
        <div
          class="undefined"
        >
          <div
            class="undefined"
          >
            <div
              class="first-column"
            />
            <div
              class="makeStyles-header-4"
              style="max-width: 100%;"
            >
              <div>
                <img
                  alt="table header"
                  data-testid="test-logo"
                  loading="lazy"
                  src="https://vsgstoqarg-539670-cdn-endpoint.azureedge.net/-/jssmedia/Project/Frontier/Dotcom/Images/fiber-2-gigabit-internet/logo-frontier.png?rev=78dfde7215fe416d80e38143dcebfa57"
                />
                <h3
                  class=""
                >
                  Header 1
                </h3>
              </div>
            </div>
          </div>
          <div>
            <div
              class="undefined undefined"
              style="background-color: rgb(242, 244, 244);"
            >
              <div
                class=""
              >
                <span>
                  <div
                    class=""
                    data-testid="test-name"
                  >
                    Speed
                  </div>
                  <div
                    class="makeStyles-tooptipStyles-7"
                  >
                    <svg
                      fill="none"
                      height="14"
                      viewBox="0 0 14 14"
                      width="14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M6.33334 3.29092C5.96515 3.29092 5.66668 3.58932 5.66668 3.95742L5.66668 4.62392C5.66668 4.99202 5.96515 5.29042 6.33334 5.29042L7.00001 5.29042C7.3682 5.29042 7.66668 4.99202 7.66668 4.62392L7.66668 3.95742C7.66668 3.58932 7.3682 3.29092 7.00001 3.29092L6.33334 3.29092Z"
                        fill="currentColor"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M13.3333 7.33158C13.3333 10.6445 10.6471 13.3301 7.33334 13.3301L6.00001 13.3301C2.6863 13.3301 1.11924e-05 10.6445 1.0882e-05 7.33158L1.07571e-05 5.99858C1.04466e-05 2.6857 2.6863 7.90697e-05 6.00001 7.87994e-05L7.33334 7.86906e-05C10.6471 7.84204e-05 13.3333 2.6857 13.3333 5.99858L13.3333 7.33158ZM7.33334 11.9971L6.00001 11.9971C3.42268 11.9971 1.33334 9.90826 1.33334 7.33158L1.33334 5.99858C1.33334 3.42189 3.42268 1.33308 6.00001 1.33308L7.33334 1.33308C9.91067 1.33308 12 3.42189 12 5.99858L12 7.33158C12 9.90826 9.91067 11.9971 7.33334 11.9971Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                      <path
                        d="M7.54168 10.2892L5.8073 10.2892L5.8073 6.29017L7.54168 6.29017L7.54168 10.2892Z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </span>
              </div>
              <div
                class="makeStyles-tableRowClass-8"
                style="max-width: 100%;"
              >
                <div
                  class="undefined"
                  data-testid="test-textValue"
                >
                  <div>
                     
                    <span>
                      2000 
                    </span>
                     Mbps download 
                    <br />
                    

                    <span>
                      2000
                    </span>
                     Mbps upload
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <p
      class="makeStyles-legal-5"
    />
  </div>
</DocumentFragment>
`;
